### PR TITLE
Add EnrolCommand section in Developer Guide

### PR DESCRIPTION
**Pull request by dylkaw**
What have you done in this PR?
* Added EnrolCommand section to the Developer Guide

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -193,7 +246,7 @@ Step 4. The user now decides that adding the member was a mistake, and decides t
 ![UndoRedoVersion3](images/UndoRedoVersion3.png)
 
 <div markdown="span" class="alert alert-info">:information_source: **Note:** If the `versionPointer` is at index 0, pointing to the initial CcaCommander version, then there are no previous CcaCommander versions to restore. The `undo` command uses `Model#canUndo()` to check if this is the case. If so, it will return an error to the user rather
-than attempting to perform the undo.
+than attempting to perform the undo.[DeveloperGuide.md](..%2F..%2FDeveloperGuide.md)
```
Comment (SelwynAng) [inline docs/DeveloperGuide.md]: I am not sure what these extra words do. Do you mind explaining it?

**Review by SelwynAng — COMMENTED** [review]
LGTM aside from the small comment given

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -154,6 +154,59 @@ Classes used by multiple components are in the `seedu.ccacommander.commons` pack
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Enrol Feature
+The enrol feature allows users to link a member to an event in order to keep track of their attendance and contributions to the event.
+
+This section will explain how the enrol feature was implemented and the various design considerations when implementing the feature.
+
+#### Implementation
+The enrollment mechanism is facilitated by `EnrolCommand`. It extends `Command`.
```
Comment (andrechuakj) [inline docs/DeveloperGuide.md]: I think the correct spelling here is "enrolment"?

**Review by andrechuakj — COMMENTED** [review]
Just a spelling amendment

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (dylkaw) [reply docs/DeveloperGuide.md]: I am unsure as to what this does, it seems as though it has been automatically added.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (dylkaw) [reply docs/DeveloperGuide.md]: Fixed!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (dylkaw) [reply docs/DeveloperGuide.md]: It has been removed.

**Pull request merged**